pkg/pattern/drain/line_tokenizer.go

```diff
@@ -1,25 +1,96 @@
 package drain
 
-import "strings"
+import (
+	"strings"
+	"unicode"
+)
 
 type LineTokenizer interface {
-	Tokenize(line string) []string
-	Join(tokens []string) string
+	Tokenize(line string) ([]string, interface{})
```

Comment:
I wonder if generics would work here, just a thought. I know interface have a cost when casting for instance.